vignettes/understanding_shapr_vaeac.Rmd

```diff
@@ -0,0 +1,981 @@
+plot_several_vaeacs_VLB_IWAE(list("Vaeac early stopping" = expl_early_stopping))
+```
+
+However, we can train it further for a fixed amount of epochs if desired.
+```{r early stopping 2, cache = TRUE}
+# Make a copy of the explanation object which we are to train further.
+expl_early_stopping_train_more <- expl_early_stopping
+
+# Continue to train the vaeac model some more epochs
+expl_early_stopping_train_more$internal$parameters$vaeac <-
+  vaeac_continue_train_model(
+    explanation = expl_early_stopping_train_more,
+    epochs_new = 15,
+    training_data = x_train,
+    verbose = TRUE
+  )
+
+expl_early_stopping_train_more$internal$parameters$vaeac <-
+  vaeac_continue_train_model(
+    explanation = expl_early_stopping_train_more,
+    epochs_new = 10,
+    training_data = x_train,
+    verbose = TRUE
+  )
+
+# Look at the training and validation errors.
+plot_several_vaeacs_VLB_IWAE(list(
+  "Vaeac early stopping" = expl_early_stopping,
+  "Vaeac early stopping more epochs" = expl_early_stopping_train_more
+))
+```
+
+Can then use the new one to create the Shapley value explanations.
+```{r early stopping 3, cache = TRUE}
+# Use extra trained vaeac model to compute Shapley values again.
+expl_early_stopping_train_more <- explain(
+  model = model,
+  x_explain = x_explain,
+  x_train = x_train,
+  approach = approach,
+  prediction_zero = p0,
+  n_combinations = n_combinations,
+  n_batches = n_batches,
+  n_samples = 250,
+  seed = seed,
+  vaeac.extra_parameters = list(
+    vaeac.pretrained_vaeac_model = expl_early_stopping_train_more$internal$parameters$vaeac
+  )
+)
+```
+And then compare them and show the Shapley values.
+```{r}
+# We can compare them
+plot_MSEv_eval_crit(list(
+  "Vaeac early stopping" = expl_early_stopping,
+  "Vaeac early stopping more epochs" = expl_early_stopping_train_more
+))
+
+# We see that the Shapley values have changed
+plot_SV_several_approaches(list(
+  "Vaeac early stopping" = expl_early_stopping,
+  "Vaeac early stopping more epochs" = expl_early_stopping_train_more
+))
+```
+
+
+
+
+
+
+## Grouping of features
+```{r vaeac grouping of featurs, cache = TRUE}
+group_list <- list(A = c("Temp", "Month"), B = c("Wind", "Solar.R"))
+approach <- "vaeac"
+n_combinations <- NULL
+n_batches <- 2
+expl_group <- explain(
+  model = model,
+  x_explain = x_explain,
+  x_train = x_train,
+  approach = "vaeac",
+  prediction_zero = p0,
+  group = group_list,
+  n_combinations = n_combinations,
+  n_batches = n_batches,
+  n_samples = n_samples,
+  keep_samp_for_vS = keep_samp_for_vS,
+  vaeac.epochs = 10,
+  vaeac.num_vaeacs_initiate = 3,
+  vaeac.verbose = TRUE
+)
+# expl_group$internal$objects
+
+# Printing the Shapley values for the test data.
+# For more information about the interpretation of the values in the table, see ?shapr::explain.
+print(expl_group$shapley_values)
+
+# Finally we plot the resulting explanations
+plot(expl_group)
+
+expl_group$internal$objects$X
+```
+
+
+
+## Mixed Data
```

Comment:
Let discuss whether we should include this or not. I did not check, but assume this by far the most time consuming part of the vignette since a bigger model is trained for longer. IF we decide to include it we should include the fitting time, and also have a very brief conclusion/discussion in the end where the performance, differences in the Shapley value and timing are compared. I guess the conclusion (for this specific use case) would then be that if you properly tune a big vaeac model you get a better-performing model than both the smaller vaeac model and the ctree model with default parameters. It should be mentioned that ctree also model parameters that can influence it's performance.

Reply:
For me it does not matter if we include it or not. I have mainly used it to check that `vaeac` works and that is give approx same values as ctree.